### ISSUE TITLE
Requests start slowing down when launching many at once

### ISSUE THREAD

**Issue by dfdx** [JuliaWeb/Requests.jl#61]
Not sure how much it's an issue with Requests.jl, but it seems to be the only package that I can reproduce it with. 

First of all, test code: 

```
using Requests
using URIParser
using Gadfly

function fetch_urls(urls; timeout=Inf)
    statuses = -ones(length(urls))
    times = -ones(length(urls))
    @sync for i=1:length(urls)
        @async begin
            try
                t = @elapsed resp = get(urls[i]; timeout=timeout)
                statuses[i] = resp.status
                times[i] = t
                print("OK\n")
            catch
                print("Error\n")
            end
        end
    end
    return (statuses, times)
end

function main()
    @time urls = [strip(l) for l in
                  open(readlines, expanduser("~/Dropbox/ips.txt"))][1:100]
    @time (statuses, times) = fetch_urls(urls)
    plot(x=[i for i=1:length(times)], y=times)
end
```

This code uses current state of master of Requests.jl and file "ips.txt" available [here](https://www.dropbox.com/s/lpp7ozj67rh26lc/ips.txt?dl=0). 

Basically, here we take 100 URLs with already resolved IPs and try to fetch them, each in a separate task. Then we collect response statuses and request run times for analysis, and visualize later in Gadfly. 

When I run this code, about 90% of requests get executed very quickly, altogether taking about 1 second. Obviously, even though Julia code is single-threaded, IO operations are executed in parallel for these requests. 

However there's some little number of slow requests, each taking 5-15 seconds (and these are actually slow URLs, so even 15 seconds is believable here): 

![request-times](https://cloud.githubusercontent.com/assets/428171/9478862/fc37332a-4b82-11e5-9c41-c3259fa40ed2.png)

But the most interesting thing is that the whole function invocation (together with slow requests) takes more than 2 minutes! So it looks like fast requests are being executed in parallel, while slow ones start executing one by one or close to it. 

I also added timeout by specifying `timeout=2` in `fetch_urls`. As expected, long requests are truncated (-1 in the graph): 

![request-timeouts](https://cloud.githubusercontent.com/assets/428171/9479210/0b05c0fe-4b85-11e5-9d72-7517e09c466c.png)

But total time is still 2 minutes! 

So I don't know whether problem is in Requests.jl, Julia tasks or just my code, but any suggestions why it takes so much time are welcome.

**Comment by sbromberger** [comment]
What's interesting to me is that the "slow" responses are fairly evenly spaced. What happens if you increase the number of requests?

**Comment by aviks** [comment]
How many `uv` threads do you have. One possiblity is that which the Julia tasks clearly do not block, maybe the jobs are being scheduled on a limited number of `uv` threads which do block? I don't know if that is the case, but it would explain the behaviour.

**Comment by dfdx** [comment]
@aviks Setting `UV_THREADPOOL_SIZE` to 32 changed nothing. Also I noticed that the time is always in 127-129 second interval, which is pretty little variance. 

@sbromberger I'll do these tests later today and come back with results.

**Comment by dfdx** [comment]
With 500 requests it took same 127 seconds, timing plot looks like this: 

![req500](https://cloud.githubusercontent.com/assets/428171/9497076/0fb64b7c-4c1c-11e5-90ee-b04483894740.png)

Removing bad performance - all URLs that took more than 2 seconds first time - doesn't look to help: total time is again 127 seconds, plot looks like this: 

![reqs500_no_bad_performers](https://cloud.githubusercontent.com/assets/428171/9497141/4fe89560-4c1c-11e5-8bed-0b95704e352a.png)

So there seems to be something special about 127(+-1) seconds.

**Comment by dfdx** [comment]
I think I mislead you - when I took another range of IPs, processing took only 3 seconds. Here's a log of latest launch on my machine (in format: time: status (id of task)):

```
!julia>     @time (statuses, times) = fetch_urls(urls, timeout=2)                                                                                                       
 29-Aug 00:27:27:INFO:root:OK (3)
 29-Aug 00:27:27:INFO:root:OK (8)
 29-Aug 00:27:27:INFO:root:OK (38)
 29-Aug 00:27:27:INFO:root:OK (65)
 29-Aug 00:27:27:INFO:root:OK (39)
 29-Aug 00:27:27:INFO:root:OK (49)
 29-Aug 00:27:27:INFO:root:OK (5)
 29-Aug 00:27:27:INFO:root:OK (4)
 29-Aug 00:27:27:INFO:root:OK (58)
 29-Aug 00:27:27:INFO:root:OK (56)
 29-Aug 00:27:27:INFO:root:OK (40)
 29-Aug 00:27:27:INFO:root:OK (7)
 29-Aug 00:27:27:INFO:root:OK (33)
 29-Aug 00:27:27:INFO:root:OK (46)
 29-Aug 00:27:27:INFO:root:OK (42)
 29-Aug 00:27:27:INFO:root:OK (67)
 29-Aug 00:27:27:INFO:root:OK (14)
 29-Aug 00:27:27:INFO:root:OK (53)
 29-Aug 00:27:27:INFO:root:OK (88)
 29-Aug 00:27:27:INFO:root:OK (15)
 29-Aug 00:27:27:INFO:root:OK (95)
 29-Aug 00:27:27:INFO:root:OK (47)
 29-Aug 00:27:27:INFO:root:OK (13)
 29-Aug 00:27:27:INFO:root:OK (91)
 29-Aug 00:27:27:INFO:root:OK (23)
 29-Aug 00:27:27:INFO:root:OK (64)
 29-Aug 00:27:27:INFO:root:OK (25)
 29-Aug 00:27:27:INFO:root:OK (60)
 29-Aug 00:27:27:INFO:root:OK (56)
 29-Aug 00:27:27:INFO:root:OK (40)
 29-Aug 00:27:27:INFO:root:OK (7)
 29-Aug 00:27:27:INFO:root:OK (33)
 29-Aug 00:27:27:INFO:root:OK (46)
 29-Aug 00:27:27:INFO:root:OK (42)
 29-Aug 00:27:27:INFO:root:OK (67)
 29-Aug 00:27:27:INFO:root:OK (14)
 29-Aug 00:27:27:INFO:root:OK (53)
 29-Aug 00:27:27:INFO:root:OK (88)
 29-Aug 00:27:27:INFO:root:OK (15)
 29-Aug 00:27:27:INFO:root:OK (95)
 29-Aug 00:27:27:INFO:root:OK (47)
 29-Aug 00:27:27:INFO:root:OK (13)
 29-Aug 00:27:27:INFO:root:OK (91)
 29-Aug 00:27:27:INFO:root:OK (23)
 29-Aug 00:27:27:INFO:root:OK (64)
 29-Aug 00:27:27:INFO:root:OK (25)
 29-Aug 00:27:27:INFO:root:OK (60)
 29-Aug 00:27:27:INFO:root:OK (82)
 29-Aug 00:27:27:INFO:root:OK (80)
 29-Aug 00:27:27:INFO:root:OK (97)
 29-Aug 00:27:27:INFO:root:OK (101)
 29-Aug 00:27:27:INFO:root:OK (57)
 29-Aug 00:27:27:INFO:root:OK (79)
 29-Aug 00:27:27:INFO:root:OK (48)
 29-Aug 00:27:27:INFO:root:OK (96)
 29-Aug 00:27:27:INFO:root:OK (69)
 29-Aug 00:27:27:INFO:root:OK (9)
 29-Aug 00:27:27:INFO:root:OK (77)
 29-Aug 00:27:27:INFO:root:OK (2)
 29-Aug 00:27:27:INFO:root:OK (11)
 29-Aug 00:27:27:INFO:root:OK (100)
 29-Aug 00:27:27:INFO:root:OK (30)
 29-Aug 00:27:27:INFO:root:OK (21)
 29-Aug 00:27:27:INFO:root:OK (35)
 29-Aug 00:27:27:INFO:root:OK (85)
 29-Aug 00:27:27:INFO:root:OK (78)
 29-Aug 00:27:27:INFO:root:OK (75)
 29-Aug 00:27:27:INFO:root:OK (80)
 29-Aug 00:27:27:INFO:root:OK (97)
 29-Aug 00:27:27:INFO:root:OK (101)
 29-Aug 00:27:27:INFO:root:OK (57)
 29-Aug 00:27:27:INFO:root:OK (79)
 29-Aug 00:27:27:INFO:root:OK (48)
 29-Aug 00:27:27:INFO:root:OK (96)
 29-Aug 00:27:27:INFO:root:OK (69)
 29-Aug 00:27:27:INFO:root:OK (9)
 29-Aug 00:27:27:INFO:root:OK (77)
 29-Aug 00:27:27:INFO:root:OK (2)
 29-Aug 00:27:27:INFO:root:OK (11)
 29-Aug 00:27:27:INFO:root:OK (100)
 29-Aug 00:27:27:INFO:root:OK (30)
 29-Aug 00:27:27:INFO:root:OK (21)
 29-Aug 00:27:27:INFO:root:OK (35)
 29-Aug 00:27:27:INFO:root:OK (85)
 29-Aug 00:27:27:INFO:root:OK (78)
 29-Aug 00:27:27:INFO:root:OK (75)
 29-Aug 00:27:27:INFO:root:OK (24)
 29-Aug 00:27:27:INFO:root:OK (1)
 29-Aug 00:27:27:INFO:root:OK (93)
 29-Aug 00:27:27:INFO:root:OK (86)
 29-Aug 00:27:27:INFO:root:OK (36)
 29-Aug 00:27:27:INFO:root:OK (31)
 29-Aug 00:27:27:INFO:root:OK (51)
 29-Aug 00:27:27:INFO:root:OK (99)
 29-Aug 00:27:28:INFO:root:OK (54)
 29-Aug 00:27:28:INFO:root:OK (68)
 29-Aug 00:27:28:INFO:root:OK (41)
 29-Aug 00:27:28:INFO:root:OK (83)
 29-Aug 00:27:28:INFO:root:OK (66)
 29-Aug 00:27:28:INFO:root:OK (50)
 29-Aug 00:27:28:INFO:root:OK (29)
 29-Aug 00:27:28:INFO:root:OK (87)
 29-Aug 00:27:28:INFO:root:OK (22)
 29-Aug 00:27:28:INFO:root:OK (72)
 29-Aug 00:27:28:INFO:root:OK (20)
 29-Aug 00:27:28:INFO:root:OK (12)
 29-Aug 00:27:28:INFO:root:OK (61)
 29-Aug 00:27:28:INFO:root:OK (52)
 29-Aug 00:27:28:INFO:root:OK (63)
 29-Aug 00:27:28:INFO:root:OK (45)
 29-Aug 00:27:28:INFO:root:OK (28)
 29-Aug 00:27:28:INFO:root:OK (81)
 29-Aug 00:27:28:INFO:root:OK (16)
 29-Aug 00:27:28:INFO:root:OK (44)
 29-Aug 00:27:28:INFO:root:OK (76)
 29-Aug 00:27:28:INFO:root:OK (37)
 29-Aug 00:27:28:INFO:root:OK (62)
 29-Aug 00:27:28:INFO:root:OK (92)
 29-Aug 00:27:28:INFO:root:OK (17)
 29-Aug 00:27:28:INFO:root:OK (94)
 29-Aug 00:27:29:WARNING:root:Error (71)
 29-Aug 00:27:29:WARNING:root:Error (55)
 29-Aug 00:27:29:WARNING:root:Error (18)
 29-Aug 00:27:29:WARNING:root:Error (26)
 29-Aug 00:27:29:WARNING:root:Error (32)
 29-Aug 00:27:29:WARNING:root:Error (59)
 29-Aug 00:27:29:WARNING:root:Error (98)
 29-Aug 00:27:29:WARNING:root:Error (34)
 29-Aug 00:27:29:WARNING:root:Error (10)
 29-Aug 00:27:29:WARNING:root:Error (74)
 29-Aug 00:27:29:WARNING:root:Error (89)
 29-Aug 00:27:29:WARNING:root:Error (73)
 29-Aug 00:27:29:WARNING:root:Error (90)
 29-Aug 00:27:29:WARNING:root:Error (43)
 29-Aug 00:27:29:INFO:root:OK (70)
 29-Aug 00:27:29:WARNING:root:Error (84)
 29-Aug 00:27:29:INFO:root:OK (19)
 29-Aug 00:27:29:INFO:root:OK (27)
 29-Aug 00:27:29:WARNING:root:Error (6)
 elapsed time: 3.015073511 seconds (36084212 bytes allocated)
```

And distribution of requests durations look like this: 

![reqs1500-1600](https://cloud.githubusercontent.com/assets/428171/9557732/16690654-4de7-11e5-95ec-2c8cfddf8ec4.png)
